syntax = "proto2";

package detection.protos;

message ImageResizer {
  oneof image_resizer_oneof {
    KeepAspectRatioResizer keep_aspect_ratio_resizer = 1;
    FixedShapeResizer fixed_shape_resizer = 2;
  }
}

enum ResizeType {
  BILINEAR = 0;
  NEAREST_NEIGHBOR = 1;
  BICUBIC = 2;
  AREA = 3;
}

message FixedShapeResizer {
  optional int32 height = 1 [default = 300];

  optional int32 width = 2 [default = 300];

  optional ResizeType resize_method = 3 [default = BILINEAR];
}

message KeepAspectRatioResizer {
  optional int32 min_dimension = 1 [default = 600];

  optional int32 max_dimension = 2 [default = 1024]; 

  optional ResizeType resize_method = 3 [default = BILINEAR];
}

